python/lsst/dm/rucio/register/rucio_interface.py

```diff
@@ -138,7 +138,44 @@ def _add_replicas(self, bundles: list[ResourceBundle]) -> None:
             A list of ResourceBundles
         """
         dids = [bundle.get_did() for bundle in bundles]
-        self.replica_client.add_replicas(rse=self.rse, files=dids)
+        retries = 0
```

Comment:
It would be nice to pull this out into a reusable function.  I had something like this:
```
def retry(func, *args, **kwargs):
    t_backoff = 2.0 + random.uniform(-0.2, 0.2)
    t_multiplier = 2.0 + random.uniform(-0.2, 0.2)
    while t_backoff < 3600 * t_multiplier:
        try:
            return func(*args, **kwargs)
        except rucio.common.exception.ServerConnectionException:
            time.sleep(t_backoff)
            t_backoff *= t_multiplier
```

I also think that retrying every `RucioException` is excessive; many of them are permanent problems.  Only retrying server problems will I think simplify some of the `try: except:` logic below, and it will fail faster when there's a real problem.

Reply:
Can you use the `backoff` package like we do in `ResourcePath`?

Reply:
I think we'd have to wrap the Rucio client functions in another level of call that we could decorate to do that, as we can't decorate the client functions themselves.